exonum/src/encoding/float.rs

```diff
@@ -66,9 +66,10 @@ impl F32 {
     /// assert!(val.is_none());
     /// ```
     pub fn try_from(value: f32) -> Option<Self> {
-        match value.classify() {
-            FpCategory::Normal | FpCategory::Zero => Some(Self { value }),
-            _ => None,
+        if value.is_normal() || value == 0.0 {
```

Comment:
Hm, at least according to my understanding of IEEE754, `+0.0 == -0.0`, though their bit-patterns are different. So it seems to me that with this code will allow both `-0.0` and `+0.0` as an internal representation of F32, potentially causing non-canonical serialization.

That is, I think the following version should be more correct:

```
if value == 0.0 { 
  Some(Self { value: 0.0 } // normalize -0.0
else if value.is_normal() {
  Some(Self { value })
else {
  None
}
```

It would be great to have a test that checks that +0.0 and -0.0 are both accepted and serialized the same in both binary and JSON format.